src/data_hub/dataframe.py

```diff
@@ -228,3 +268,57 @@ def _constructor(self):
         """this method overrides the _constructor property to return a DataFrame and is required for compatibility with a pandas DataFrame"""
 
         return DataFrame
+
+
+@beartype
+def _infer_column_type(column: pd.Series):
+    """utility function to infer type of data in a pandas column"""
+
+    non_null_values = column.dropna()
+
+    if non_null_values.empty:
```

Comment:
Is it possible to defer the creation of the column until inference could at least make a reasonable guess? If it's empty, then the default could very likely be wrong.

Reply:
column is created after inference. (we can't create a column without knowing what type...)

the failure mode of the inference code is to return `string`